packages/config/src/index.ts

```diff
@@ -186,12 +186,25 @@ export default async (
     'workspace-concurrency': 4,
     'workspace-prefix': opts.workspaceDir,
   })
+
+  const settingKeys = Object.keys(npmConfig?.sources?.project?.data ?? {})
+  const rcOptions = Object.keys(rcOptionsTypes)
```

Comment:
There's an issue with this solution because the rcOptionsTypes that is passed to the config only lists the options related to the current command:

https://github.com/pnpm/pnpm/blob/6f1254166287fc2db155fc7ee4cf274b51dc9ebb/packages/pnpm/src/main.ts#L94

We need somehow to post the rc option types of all commands. Might be OK just to merge the rc options of all plugins and pass them to the config funcion